### DISCUSSION TITLE
Custom plugin extending react-apollo

### DISCUSSION THREAD

**Discussion by correttojs** [dotansimha/graphql-code-generator#4229] [General] [answered]
Hi,
I'm trying to create a plugin to autogenerate the read and write operations to the Apollo InMemoryCache.

I pushed some sample code based on the react-apollo plugin here: https://github.com/correttojs/graphql-codegen-apollo-cache

Once I tested the plugin together with the react-apollo plugin, all the query documents were duplicated. 
so I added  
`config.documentMode = DocumentMode.external;`

and I removed the Operations import: 
`prepend: visitor.getImports().filter((i) => !/Operations/.test(i)),`

I also wanted to generate my new plugin into a different file (in order to avoid some circular dependencies eslint warnings). So I created a second output definition within the codegen.yml configuration and included all the imports to the required query documents and types.

My question is: is there a better way to achieve the same result?

Would you be interested in this plugin? I could complete it following your suggestions

**Comment by dotansimha** [comment, marked as answer]
Hi @correttojs 

You plugin extends `ClientSideBaseVisitor` which already generates documents for you. I think maybe we need to break it down to small pieces (the `buildOperation`) method, and allow you to skip generating documents again. 

PRs are welcome ;)

**Reply by correttojs** [reply]
I need to configure the  `import-types` preset. I added an example to my repo
